src/modules/database.ts

```diff
@@ -1,16 +1,133 @@
 import {CMConfig} from 'app/config';
 import {RestClient} from 'app/client';
-import {FindRequest, FindResponse} from 'app/types/codemash.dtos';
+import {
+  AggregateRequest,
+  CountRequest,
+  DeleteManyRequest,
+  DeleteOneRequest,
+	DistinctRequest,
+	FindOneRequest,
+	FindRequest,
+	FindTermsChildrenRequest,
+	FindTermsRequest,
+	InsertManyRequest,
+	InsertOneRequest,
+  ReplaceOneRequest,
+  ResponseStatus,
+  UpdateManyRequest,
+  UpdateOneRequest,
+} from 'app/types/codemash.dtos';
+import {ApiResult} from '@servicestack/client';
 
-export async function getRecords(
+const isJsonResponse = (value: any): value is string => typeof value === 'string';
+interface ICMDbResult {
+  response: any;
+  responseStatus?: ResponseStatus;
+  isSuccess: boolean;
+  isError: boolean;
+}
+
+const transformApiResult = (target: ApiResult<any>): ICMDbResult => {
+  // transform given api string result to a js object.
+  const { isSuccess, response, isError } = target;
+  const { responseStatus, result } = response || {};
+
+  return {
+    isSuccess,
+    isError,
+    responseStatus,
+    response: isJsonResponse(result) ? JSON.parse(result) : response
+  }
+}
+
+export async function findMany(
 	request: FindRequest
-): Promise<FindResponse | undefined> {
-	var client = RestClient.Json(new CMConfig());
+) {
+	const client = RestClient.Json(new CMConfig());
```

Comment:
Butu gerai gal config ikisti kazkaip i Singleton. Ka manai ? https://medium.com/geekculture/how-to-implement-singleton-in-typescript-3af39070d01